rpi/gpio.go

```diff
@@ -59,20 +95,49 @@ func (gp gpioPin) SetPWMFreq(ctx context.Context, freqHz uint, extra map[string]
 	return gp.pi.SetPWMFreqBcom(gp.bcom, freqHz)
 }
 
+func (pi *piPigpio) reconfigureGPIOs(ctx context.Context, cfg *Config) error {
+	// Set new pins based on config
+	pi.gpioPins = map[int]*rpiGPIO{}
+	for _, newConfig := range cfg.Pins {
+		if newConfig.Type != rpiutils.PinGPIO {
+			continue
+		}
+		bcom, have := rpiutils.BroadcomPinFromHardwareLabel(newConfig.Pin)
+		if !have {
+			return errors.Errorf("no hw pin for (%s)", newConfig.Pin)
+		}
+		pin := &rpiGPIO{name: newConfig.Name, pin: bcom}
+		pi.gpioPins[int(bcom)] = pin
+		pi.logger.Warn("yo pin: ", pin.name)
+	}
+	return nil
+}
+
 // GetGPIOBcom gets the level of the given broadcom pin
 func (pi *piPigpio) GetGPIOBcom(bcom int) (bool, error) {
 	pi.mu.Lock()
 	defer pi.mu.Unlock()
-	if !pi.gpioConfigSet[bcom] {
-		if pi.gpioConfigSet == nil {
-			pi.gpioConfigSet = map[int]bool{}
+
+	// verify we are currently managing this pin via GPIOPinByName or reconfigure
+	pin, ok := pi.gpioPins[bcom]
+	if !ok {
+		return false, fmt.Errorf("error getting GPIO pin, pin %v not found", bcom)
+	}
+	// configure the pin to be an input if it is not already an input
+	if pin.configuration != GPIOInput {
+		// first if the pin was configured for pwm, we should turn off the pwm
+		if pin.configuration == GPIOPWM {
+			res := C.set_PWM_dutycycle(pi.piID, C.uint(pin.pin), C.uint(0))
+			if res != 0 {
+				return false, errors.Errorf("pwm set fail %d", res)
+			}
```

Comment:
currently I turn off the pwm if a user tries to read the state of the pin, since reading the state of the pin is kinda useless if you are using a pwm. 

This is a behavior change so I am okay with removing it

Reply:
Since pigpio allows you to do this, I think we should keep the old behaviour.

Reply:
alright, will move things around to track whether a PWM is configured separately so digital outputs can work